src/python/pants/core/goals/generate_lockfiles.py

```diff
@@ -45,10 +56,15 @@ class GenerateLockfile:
 
     Subclasses will usually want to add additional properties, such as what requirements to
     install and Python interpreter constraints.
+
+    Always set `diff=False` when creating instances, it will be mutated to `True` in case lockfile
```

Comment:
This comment seems more confusing than useful. It's not an intrinsic property of this class that this must be set to False. It's more that in practice we always set it to False and then set it to True when we need to. But that's nothing to do with this class's definition.

Reply:
If there's a better way to phrase it, I'm all for that. I wanted to document someewhere precisely what you stated that "we always set it to False and then set it to True when we need to" in way that is somewhat discoverable. I don't understand what you mean with that this is not an intrinsic property of the class. It is a property of how the class is supposed to be used, and this is the docstring for the class. 🤷🏽  (i.e. your comment does not give me enough guidance as to what corrective action you think would be acceptable here)

Reply:
I guess I don't see why corrective action is needed? If someone wants the diff they will set this to True, if they don't they will set it to False. We happen to use this in a certain way in practice, currently, but it doesn't seem like a requirement of the class itself that it must be used this way?  What is the error this is trying to prevent?

Reply:
> I guess I don't see why corrective action is needed?

Maybe badly phrased by me. By this I meant what I should change in the docstring.. ;) 

> If someone wants the diff they will set this to True.

Which won't work.

> What is the error this is trying to prevent?

The above mistake ;) 

If you set `diff=True`, and indeed, even provide a `diff` object in the result, it still won't be printed as long as the `diff` option is false. 

Perhaps we should only check the subsystem option once, when we set `diff=True` on all requests, and then print all diff results we get regardless?

Reply:
Changed it, so that any `GenerateLockfileResult` that has a `diff` will be printed. So if there's a backend that sets `diff=True` on its `GenerateLockfile` request, it will be printed regardless of the `diff` option on the `generate-lockfiles` subsystem.

Reply:
It won't be printed, but it will be returned in the result, no? Which is the contract of this class as far as I can tell.

Reply:
Yes.